exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -285,82 +284,82 @@ def _export(
-                            "google.rpc.retryinfo-bin"
-                        )
-                        if retry_info_bin is not None:
-                            retry_info = RetryInfo()
-                            retry_info.ParseFromString(retry_info_bin)
-                            delay = (
-                                retry_info.retry_delay.seconds
-                                + retry_info.retry_delay.nanos / 1.0e9
-                            )
-
-                        logger.warning(
-                            (
-                                "Transient error %s encountered while exporting "
-                                "%s to %s, retrying in %ss."
-                            ),
-                            error.code(),
-                            self._exporting,
-                            self._endpoint,
-                            delay,
-                        )
-                        sleep(delay)
-                        continue
-                    else:
-                        logger.error(
-                            "Failed to export %s to %s, error code: %s",
-                            self._exporting,
-                            self._endpoint,
-                            error.code(),
-                            exc_info=error.code() == StatusCode.UNKNOWN,
+                    if retry_info_bin is not None:
+                        retry_info = RetryInfo()
+                        retry_info.ParseFromString(retry_info_bin)
+                        delay = (
+                            retry_info.retry_delay.seconds
+                            + retry_info.retry_delay.nanos / 1.0e9
                         )
 
+                    logger.warning(
+                        (
+                            "Transient error %s encountered while exporting "
+                            "%s to %s, retrying in %ss."
+                        ),
+                        error.code(),
+                        self._exporting,
+                        self._endpoint,
+                        delay,
+                    )
+                    self._shutdown_occuring.wait(delay)
+                    continue
+                else:
+                    # Should not be possible ?
                     if error.code() == StatusCode.OK:
                         return self._result.SUCCESS
+                    logger.error(
+                        "Failed to export %s to %s, error code: %s",
+                        self._exporting,
+                        self._endpoint,
+                        error.code(),
+                        exc_info=error.code() == StatusCode.UNKNOWN,
+                    )
 
                     return self._result.FAILURE
 
         return self._result.FAILURE
 
     def shutdown(self, timeout_millis: float = 30_000, **kwargs) -> None:
-        if self._shutdown:
+        if self._shutdown_occuring.is_set():
```

Comment:
This already had the same problem, but `shutdown()` is not thread safe. I guess for this PR we can assume only one thread calls it.